### PR TITLE
MacOS debug symbols support

### PR DESCRIPTION
**Pull request by andrew-m-leonard**
Enable jdk8 support for MacOS dSYM debug symbols

Signed-off-by: Andrew Leonard <anleonar@redhat.com>

### CODE REVIEW

**Comment by karianna** [comment]
@andrew-m-leonard has this been submitted upstream?

**Comment by andrew-m-leonard** [comment]
@karianna not yet.... which way would you like to play this?

**Comment by karianna** [comment]
These changes have all been manually tested I assume?  I think in that case you could write into the jdk8u-dev mailing list and say something like _"We've been working on this functionality at Adopt, we'd like to add it to 8u, we have a working patch that's tested (paste the contents this PR) but we're looking for advice on whether A.) There's any appetite to add this functionality to 8 and B.) If so have we taken the right approach?"_

I think we should go over this PR in a call as well.  I'll ping you on Slack

**Comment by andrew-m-leonard** [comment]
@karianna yes all tested on jdk8 mac. I want to re-test my mac only jdk8 PR now I have taken AIX out of it

File: make/common/NativeCompilation.gmk [hunk 1]
```diff
@@ -490,7 +499,14 @@ define SetupNativeCompilation
 			$(CD) $$(@D) && $(OBJCOPY) --add-gnu-debuglink=$$(@F) $$<
 			$(TOUCH) $$@
             endif
-          endif # No MacOS X support
+          else ifeq ($(OPENJDK_TARGET_OS), macosx)
+            ifneq ($$($1_STRIP_POLICY), no_strip)
+              $1_DEBUGINFO_FILES := $$($1_OBJECT_DIR)/$$($1_BASENAME).dSYM/Contents/Info.plist \
+                $$($1_OBJECT_DIR)/$$($1_BASENAME).dSYM/Contents/Resources/DWARF/$$($1_BASENAME)
+              $$($1_DEBUGINFO_FILES): $$($1_TARGET)
+			$(DSYMUTIL) --out $$($1_OBJECT_DIR)/$$($1_BASENAME).dSYM $$<
```
Comment (karianna) [inline make/common/NativeCompilation.gmk]: whitespace is odd here?  No biggie.

File: make/common/NativeCompilation.gmk [hunk 2]
```diff
@@ -508,8 +524,7 @@ define SetupNativeCompilation
               $1 += $$(subst $$($1_OBJECT_DIR),$$($1_OUTPUT_DIR),$$($1_DEBUGINFO_FILES))
             endif
           endif
-        endif
-      endif # !MacOS X
+      endif # ENABLE_DEBUG_SYMBOLS
```
Comment (karianna) [inline make/common/NativeCompilation.gmk:527]: Something wrong about the whitespace (4 vs 2) here which makes me wonder if we've missed indenting/scoping above

Comment (andrew-m-leonard) [reply make/common/NativeCompilation.gmk:527]: yes, I didn't want to change the whole block indent, but I can do that

File: make/common/NativeCompilation.gmk (hunk 1 above)
Comment (andrew-m-leonard) [reply make/common/NativeCompilation.gmk]: the target cmds have to be "tab"

**Comment by andrew-m-leonard** [comment]
@karianna i've sorted out the indenting, makes it a bit more tricky to review, but end result is now consistent.
Note the importance of make cmds being "tab" indented that is very important

**Comment by andrew-m-leonard** [comment]
Test build with latest update: https://ci.adoptopenjdk.net/view/Failing%20Builds/job/build-scripts/job/jobs/job/jdk8u/job/jdk8u-mac-x64-hotspot/938/

File: make/common/NativeCompilation.gmk [hunk 3]
```diff
@@ -440,76 +440,91 @@ define SetupNativeCompilation
   ifeq ($$($1_STATIC_LIBRARY),)
     ifneq ($$($1_DEBUG_SYMBOLS),)
       ifeq ($(ENABLE_DEBUG_SYMBOLS), true)
-        ifneq ($(OPENJDK_TARGET_OS), macosx) # no MacOS X support yet
-          ifneq ($$($1_OUTPUT_DIR),$$($1_OBJECT_DIR))
+        ifneq ($$($1_OUTPUT_DIR),$$($1_OBJECT_DIR))
+          ifeq ($(OPENJDK_TARGET_OS), macosx)
+	    # MacOS debug symbols are in %.dSYM directories
+	    $$($1_OUTPUT_DIR)/$$($1_BASENAME).dSYM/Contents/Info.plist : $$($1_OBJECT_DIR)/$$($1_BASENAME).dSYM/Contents/Info.plist
```
Comment (karianna) [inline make/common/NativeCompilation.gmk]: Does this need indenting?

**Review by karianna — CHANGES_REQUESTED** [review]
One more indent nitpick

File: make/common/NativeCompilation.gmk (hunk 3 above)
Comment (andrew-m-leonard) [reply make/common/NativeCompilation.gmk]: no, the make target:dependency rule line is space indented, it is just the recipe lines that must be tabs

Comment (andrew-m-leonard) [reply make/common/NativeCompilation.gmk]: ah there is a tab on that line, will remove

**Comment by andrew-m-leonard** [comment]
Hmm, that indent has broken it! need to double check tabs/spaces again!

**Comment by andrew-m-leonard** [comment]
Good now: https://ci.adoptopenjdk.net/job/build-scripts/job/jobs/job/jdk8u/job/jdk8u-mac-x64-hotspot/940/

**Review by karianna — APPROVED** [review]

**Comment by karianna** [comment]
@andrew-m-leonard you may need to PR into the release branch as well.

**Comment by andrew-m-leonard** [comment]
@karianna thanks for reminding me

**Comment by andrew-m-leonard** [comment]
Merged upstream into jdk8u302: https://hg.openjdk.java.net/jdk8u/jdk8u-dev/rev/af65dfd2bea4

**Pull request merged**